lib/parallel_tests/cli.rb

```diff
@@ -107,6 +108,19 @@ def run_tests_in_parallel(num_processes, options)
       end
     end
 
+    def env_index(options)
+      return @env_index if defined?(@env_index)
+
+      start = options[:first_is_1] ? 0 : -1
+
+      @env_index =
+        if options[:allow_duplicates]
+          proc { start += 1 }
+        else
+          ->(groups, group) { groups.index(group) }
```

Comment:
This is the original behavior. As I was writing this, I was wondering:

1. Could we use the proc/counter as a default for everything?
2. Is there something else I'm missing like "spec order randomization" where it's not as simple as just using an incrementing counter (e.g. maybe we don't even need the proc; we could just increment)

If it's not as simple as this proc (or a basic counter) I'm open to other suggestions; the only other idea I had was to track the index in a hash with a counter so I could see when I've re-used a group already.

If none of these ideas would work, then maybe `--allow-duplicates` should only work with a single file? This would be OK for my primary use case (and still useful I think)